pkg/actuators/machineset/ec2_instance_types.go

```diff
@@ -179,3 +186,20 @@ func getGpuCount(gpuInfo *ec2.GpuInfo) int64 {
 	}
 	return gpuCountSum
 }
+
+// normalizeArchitecture converts the given architecture string from the format used by the EC2 API to the one for kubernetes.
+// In particular, at the time of writing,
+// the EC2 API uses the GNU name for the x86_64 architecture, and the Golang/LLVM name for the aarch64.
+// The kubernetes.io/arch label expects the Golang/LLVM names.
+// See vendor/github.com/aws/aws-sdk-go/service/ec2/api.go
+func normalizeArchitecture(architecture string) string {
+	switch architecture {
+	case ec2.ArchitectureTypeX8664:
+		return "amd64"
+	case ec2.ArchitectureTypeArm64:
+		return "arm64"
+	}
+	klog.V(2).Infof("unknown architecture %s. Defaulting to amd64", architecture)
+	// Default to amd64 if we don't recognize the architecture.
+	return "amd64"
```

Comment:
i think it would be nice to make these values constants just to reduce the possibility of differences.

Reply:
added them as constants of a dedicated type,

```golang
type normalizedArch string

const (
	ArchitectureAmd64 normalizedArch = "amd64"
	ArchitectureArm64 normalizedArch = "arm64"
)
```

Reply:
wdyt?